### PR TITLE
more determinism in delete order (forward port of #16401)

### PR DESCRIPTION
**Pull request by BryceStansfieldAt-da**
Port #16401 and #16448 from 2.3.x to main.

### CODE REVIEW

File: ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala [hunk 1]
```diff
@@ -5,14 +5,14 @@ package com.daml.http.dbbackend
...
 import nonempty.NonEmptyReturningOps._
-
 import doobie._
 import doobie.implicits._
+
 import scala.annotation.nowarn
-import scala.collection.immutable.{Seq => ISeq, SortedMap}
-import scalaz.{@@, Cord, Functor, OneAnd, Tag, \/, -\/, \/-}
+import scala.collection.immutable.{SortedMap, Seq => ISeq}
+import scalaz.{-\/, @@, Cord, Functor, OneAnd, Tag, \/, \/-}
```
Comment (S11001001) [inline ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala]: 6fc9dab2393c4f646d3108b5a66036caaca44427

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsFetch.scala [hunk 2]
```diff
@@ -295,70 +295,79 @@ private class ContractsFetch(
...
-          .via(conflation)
-          .map(insertAndDelete)
+          val transactInsertsDeletes = Flow
+            .fromFunction(
+              jsonifyInsertDeleteStep(
+                (_: InsertDeleteStep[Any, lav1.event.CreatedEvent]),
+                templateId,
+              )
+            )
+            .via(conflation)
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsFetch.scala]: https://github.com/digital-asset/daml/pull/16401/files#diff-73c6cf12a0baa5992ba41a5b30d92e9ffbe234debd5876949b5cff238de312e3R337 is needed as well. 457af4b780d06871d1a6ba685bd48733e5491dd3

File: ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala [hunk 3]
```diff
@@ -311,29 +314,34 @@ sealed abstract class Queries(tablePrefix: String, tpIdCacheMaxEntries: Long)(im
...
+              fr" OR ",
+            )
+          }
+          del.update.run
+        case _ =>
+          free.connection.pure(0)
+      }
+    } else {
+      Update[String](s"DELETE FROM $contractTableNameRaw WHERE contract_id = ?")
+        .updateMany(cids.values.flatten.toSeq.sorted)
```
Comment (S11001001) [inline ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala]: Must also include `tpid`; see #16401 checklist. d6db66fc80e0f31599e461eb19293ec2077f3132

**Review by S11001001 — APPROVED** [review]
Included suggestions as commits on the branch; @BryceStansfieldAt-da please take a look and merge if you have no followups to those.

@ray-roestenburg-da could you take a second look, as the determinism logic has necessarily changed for interface support, please?

**Review by ray-roestenburg-da — APPROVED** [review]

**Pull request merged**